channeld/channeld.c

```diff
@@ -3458,7 +3458,7 @@ static void resume_splice_negotiation(struct peer *peer,
 	u32 chan_output_index;
 	struct bitcoin_signature their_sig;
 	struct pubkey *their_pubkey;
-	struct bitcoin_tx *final_tx;
+	struct bitcoin_tx *final_tx = NULL;
```

Comment:
nit: this is not the way we usually init the value, I think that `msg_received` is a good example

Reply:
To add a bit more context: touching uninitialized memory, despite not doing something useful with that memory is an indicator of a logic error. Just initializing here would paper over a real issue, and it should only be done in extreme circumstances where that is indeed the desired behavior. In this case we never construct the final TX but then try to access it anyway? We should rather have skipped the access based on the prior decisions not to initialize it.

For reference: we almost exclusively set variables to `NULL` when looping over a set to find something, and we use the `NULL` value as a way to say "not found", hence the uninitialized state has meaning on its own.